airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/ReplicationActivityImpl.java

```diff
@@ -147,33 +171,90 @@ private static StandardSyncOutput reduceReplicationOutput(final ReplicationOutpu
                                                                                                     final IntegrationLauncherConfig destinationLauncherConfig,
                                                                                                     final JobRunConfig jobRunConfig,
                                                                                                     final StandardSyncInput syncInput) {
-    return () -> {
-      final IntegrationLauncher sourceLauncher = new AirbyteIntegrationLauncher(
-          sourceLauncherConfig.getJobId(),
-          Math.toIntExact(sourceLauncherConfig.getAttemptId()),
-          sourceLauncherConfig.getDockerImage(),
-          processFactory,
-          syncInput.getResourceRequirements());
-      final IntegrationLauncher destinationLauncher = new AirbyteIntegrationLauncher(
-          destinationLauncherConfig.getJobId(),
-          Math.toIntExact(destinationLauncherConfig.getAttemptId()),
-          destinationLauncherConfig.getDockerImage(),
-          processFactory,
-          syncInput.getResourceRequirements());
-
-      // reset jobs use an empty source to induce resetting all data in destination.
-      final AirbyteSource airbyteSource =
-          sourceLauncherConfig.getDockerImage().equals(WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB) ? new EmptyAirbyteSource()
-              : new DefaultAirbyteSource(sourceLauncher);
-
-      return new DefaultReplicationWorker(
-          jobRunConfig.getJobId(),
-          Math.toIntExact(jobRunConfig.getAttemptId()),
-          airbyteSource,
-          new NamespacingMapper(syncInput.getNamespaceDefinition(), syncInput.getNamespaceFormat(), syncInput.getPrefix()),
-          new DefaultAirbyteDestination(destinationLauncher),
-          new AirbyteMessageTracker(),
-          new AirbyteMessageTracker());
+    return () -> new Worker<>() {
+
+      final AtomicBoolean cancelled = new AtomicBoolean(false);
+      Process process;
+
+      @Override
+      public ReplicationOutput run(StandardSyncInput standardSyncInput, Path jobRoot) throws WorkerException {
+        try {
+          final Path jobPath = WorkerUtils.getJobRoot(workspaceRoot, jobRunConfig.getJobId(), jobRunConfig.getAttemptId());
+
+          final Map<String, String> fileMap = Map.of(
+              INIT_FILE_APPLICATION, REPLICATION,
+              INIT_FILE_JOB_RUN_CONFIG, Jsons.serialize(jobRunConfig),
```

Comment:
these configs can get quite large - are there any worries here?

Reply:
Yes, there is a temporal size limit. I am not sure about the exact limit, from what I remember from cadence it should be aroung 4kb,

Reply:
These files are copied directly via Kubernetes. They aren't passed via Temporal.